src/sentry/migrations/0644_backfill_priority_for_groups.py

```diff
@@ -0,0 +1,178 @@
+
+# copied to ensure migraitons work if the enums change #
+
+logger = logging.getLogger(__name__)
+
+
+class GroupSubStatus:
+    # GroupStatus.IGNORED
+    UNTIL_ESCALATING = 1
+    # Group is ignored/archived for a count/user count/duration
+    UNTIL_CONDITION_MET = 4
+    # Group is ignored/archived forever
+    FOREVER = 5
+
+    # GroupStatus.UNRESOLVED
+    ESCALATING = 2
+    ONGOING = 3
+    REGRESSED = 6
+    NEW = 7
+
+
+class PriorityLevel:
+    LOW = 25
+    MEDIUM = 50
+    HIGH = 75
+
+
+class GroupCategory(Enum):
+    ERROR = 1
+    PERFORMANCE = 2
+    PROFILE = 3  # deprecated, merging with PERFORMANCE
+    CRON = 4
+    REPLAY = 5
+    FEEDBACK = 6
+
+
+PERFORMANCE_P95_ENDPOINT_REGRESSION_GROUPTYPE_ID = 1018
+PROFILE_FUNCTION_REGRESSION_TYPE_ID = 2011
+
+
+# end copy #
+
+BATCH_SIZE = 100
+
+UPDATE_QUERY = """
+    UPDATE sentry_groupedmessage
+    SET priority = new_data.priority,
+    data = new_data.data::text
+    FROM (VALUES %s) AS new_data(id, priority, data)
+    WHERE sentry_groupedmessage.id = new_data.id AND sentry_groupedmessage.priority IS NULL
+"""
+
+REDIS_KEY = "priority_backfill.last_processed_id"
+
+
+def _get_priority_level(group_id, level, type_id, substatus):
+    group_type = get_group_type_by_type_id(type_id)
+
+    # Replay and Feedback issues are medium priority
+    if group_type.category in [GroupCategory.REPLAY.value, GroupCategory.FEEDBACK.value]:
+        return PriorityLevel.MEDIUM
+
+    # All escalating issues are high priority for all other issue categories
+    if substatus == GroupSubStatus.ESCALATING:
+        return PriorityLevel.HIGH
+
+    if group_type.category == GroupCategory.ERROR.value:
+        if level in [logging.INFO, logging.DEBUG]:
+            return PriorityLevel.LOW
+        elif level == logging.WARNING:
+            return PriorityLevel.MEDIUM
+        elif level in [logging.ERROR, logging.FATAL]:
+            return PriorityLevel.HIGH
+
+        logging.warning('Unknown log level "%s" for group %s', level, group_id)
+        return PriorityLevel.MEDIUM
+
+    if group_type.category == GroupCategory.CRON.value:
+        if level == logging.WARNING:
+            return PriorityLevel.MEDIUM
+
+        return PriorityLevel.HIGH
+
+    # Profiling issues should be treated the same as Performance issues since they are merging
+    if group_type.category in [GroupCategory.PERFORMANCE.value, GroupCategory.PROFILE.value]:
+        # Statistical detectors are medium priority
+        if type_id in [
+            PROFILE_FUNCTION_REGRESSION_TYPE_ID,
+            PERFORMANCE_P95_ENDPOINT_REGRESSION_GROUPTYPE_ID,
+        ]:
+            return PriorityLevel.MEDIUM
+        return PriorityLevel.LOW
+
+    # All other issues are the default medium priority
+    return PriorityLevel.MEDIUM
+
+
+def update_group_priority(apps, schema_editor):
+    Group = apps.get_model("sentry", "Group")
+
+    redis_client = redis.redis_clusters.get(settings.SENTRY_MONITORS_REDIS_CLUSTER)
```

Comment:
why are we using the monitors redis cluster for this?

Reply:
Using any cluster that has space since it's just one key - related: https://github.com/getsentry/sentry/pull/64677

Reply:
Yeah, it didn't feel worthwhile to try provision some other cluster for this, so I chose an arbitrary one. I think we will end up generalising this storing progress pattern, we can define a specific cluster for this then